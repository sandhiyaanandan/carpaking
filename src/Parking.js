import React from 'react';
import {TOTAL_AVAILABLE_SPACE} from './constants';
import {useHistory } from "react-router-dom";
function Parking()
{
	let parkingBox = [];
	let no_available_space = TOTAL_AVAILABLE_SPACE;
	const history = useHistory(); //used to navigate to other page or component
	var alertMsg="";
	const isParkingSelected = (e) => {	
		var cUsrDetails =null;
		let isCarParked = document.activeElement.style;
		var curUser = localStorage.getItem("currentUser");
		if(curUser !== null) {
			cUsrDetails = JSON.parse(curUser); 
		}
		if(isCarParked.backgroundColor === "" || isCarParked.backgroundColor === "white") {
			if( no_available_space > 0 ) {
				if (alertMsg === "Car In") {
					alert("Your car is already parked..Thank you");
					return;
				} else {
					alertMsg = "Car In";
					isCarParked.backgroundColor = "yellow";
				}
			}
				cUsrDetails.parkingPos =  e.target.value;
				localStorage.setItem('currentUser', JSON.stringify(cUsrDetails));
			 no_available_space--;
		} else {
			//Get carnumber from user and check in local storage...if available change color to white
			var userCarNumber  = prompt("Please enter your carNumber");
			if ( cUsrDetails.carNumber === userCarNumber )
			{
				cUsrDetails.parkingPos = -1;
				alertMsg = "Car OUT";
				isCarParked.backgroundColor = "white";
				if( no_available_space >= 0 ) {
					no_available_space++;
				}
			} else {
				alertMsg = "Try again. Please provide registered carNumber";
				history.push("/Login");
			}
		}
		alert(alertMsg);
		var regiteredDetails = localStorage.getItem("regDetails");
		let custDetails =[];
		if(regiteredDetails !== null) {
			custDetails = JSON.parse(regiteredDetails); 
			if (custDetails.length > 0 )
			{
				custDetails = custDetails.map((cDetail) => {
					if(cDetail.carNumber === cUsrDetails.carNumber) {
						return {
							...cDetail,
							parkingPos: cUsrDetails.parkingPos
						}
					}
					return cDetail;
				})
			} else {
				custDetails.push(cUsrDetails);
			}
		}
		localStorage.setItem('regDetails', JSON.stringify(custDetails));
		localStorage.setItem('no_available_space', no_available_space);
	}
	
	const allUser = localStorage.getItem("regDetails");
	var posArr=0;
	if(allUser !== null) {
		const cDetails = JSON.parse(allUser); 
		if (cDetails.length > 0 )
		{
			posArr = (cDetails.map((detail)=> {
				let pos = parseInt(detail.parkingPos);
				return pos;
			}));
		}
	}

	for( let pos = 0; pos < TOTAL_AVAILABLE_SPACE; pos++)
	{
		if( pos % 5 === 0 )
			parkingBox.push(<br/>);
		if(posArr && ( posArr.indexOf(pos)!== -1))
		{
			parkingBox.push(<input type="button" style={{backgroundColor:'yellow'}} value={pos} className="box" onClick={isParkingSelected}/>);
		} else {
			parkingBox.push(<input type="button" value={pos} className="box" onClick={isParkingSelected}/>);
		}		
	}

	return(<div style={{backgroundColor:'red'}} data-test="Parking-Page-Header">
		<h1 className="pageHeader" style={{color:'white'}}>Parking Area</h1>
		<div className="parkingArea" >
			{parkingBox}
		</div>
	</div>)	
}
export default Parking;