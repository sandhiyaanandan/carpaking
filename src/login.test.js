import React from 'react';
import { shallow, mount } from 'enzyme';
import Login from './login';
import {findByTestAttr} from './../jestutils';
import toJson from 'enzyme-to-json';
import Parking from './Parking';
const setUp =()=>{
    const component = shallow(<Login />);
    return component;
};

describe('Login', ()=>{
  let wrapper;
    beforeEach(() =>{
        wrapper = setUp();
    });

it('should render the Login Component correctly', ()=>{   
      expect(toJson(wrapper)).toMatchSnapshot();
    });

it('should render Login component correctly', ()=>{
  let comp = findByTestAttr(wrapper, 'login');
  expect(comp.length).toBe(1);
});

// make our assertion and what we expect to happen 
 it('should render loginForm without throwing an error', () => {
   expect(wrapper.find('form#loginForm').exists()).toBe(true);
 })

describe('Login input', () => { 
 it('should respond to change event and change the state of the Login name', () => {
   wrapper.find('#name').simulate('change', {target: {value: 'user'}});
   expect(wrapper.state('userName')).toEqual('user');
  })
});

describe('Password input', () => {
 it('should respond to change event and change the state of the Password', () => {
   wrapper.find('#password').simulate('change', {target: {value: 'admin'}});
   expect(wrapper.state('password')).toEqual('admin');
  })

});

/*describe('Submit details', () => {

  it('should call handleFormSubmit when form submitted via button', () => {
    window.alert = jest.fn();
   
    //const fakeEvent = () => { return true; };
    //jest.clearAllMocks();
    wrapper.find('form#loginForm').find('[className="submitButton"]').simulate('click');
     wrapper.instance().isLogin = false;
    console.log("-----isLogin---"+wrapper.instance().isLogin);
    const spy =  async () => {
      const data = await jest.spyOn(wrapper.instance(), 'isLoginStatus');
      return data;
    }
    console.log("*******dta****"+spy);
  
    window.alert.mockClear();
    }) 
});
*/

});