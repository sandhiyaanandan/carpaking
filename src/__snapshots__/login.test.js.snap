// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login Component correctly 1`] = `
<styled.div>
  <div
    data-test="login"
    style={
      Object {
        "backgroundColor": "red",
        "height": "370px",
      }
    }
  >
    <div
      className="loginBox"
    >
      <h2
        className="pageHeader"
      >
        Login
      </h2>
      <form
        id="loginForm"
      >
        <styled.input
          id="name"
          onChange={[Function]}
          placeholder="Enter Login Name"
          type="text"
        />
        <styled.input
          id="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
        />
        <styled.input
          className="submitButton"
          onClick={[Function]}
          type="button"
          value="submit"
        />
      </form>
    </div>
    <p
      style={
        Object {
          "color": "white",
        }
      }
    >
      New User? 
      <span
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "underline",
          }
        }
      >
        Registeration
      </span>
    </p>
  </div>
</styled.div>
`;
